dspace-api/src/main/java/org/dspace/statistics/export/processor/ExportEventProcessor.java

```diff
@@ -0,0 +1,288 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.statistics.export.processor;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.commons.codec.CharEncoding;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+import org.dspace.content.DCDate;
+import org.dspace.content.Entity;
+import org.dspace.content.EntityType;
+import org.dspace.content.Item;
+import org.dspace.content.MetadataValue;
+import org.dspace.content.factory.ContentServiceFactory;
+import org.dspace.content.service.EntityService;
+import org.dspace.content.service.EntityTypeService;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.dspace.core.Utils;
+import org.dspace.services.ConfigurationService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.dspace.statistics.export.factory.OpenURLTrackerLoggerServiceFactory;
+import org.dspace.statistics.export.service.OpenUrlService;
+
+/**
+ * Abstract export event processor that contains all shared logic to handle both Items and Bitstreams
+ * from the ExportUsageEventListener
+ */
+public abstract class ExportEventProcessor {
+
+    private static Logger log = Logger.getLogger(ExportEventProcessor.class);
+
+    /* The metadata field which is to be checked for */
+    protected String trackerTypeMetadataField;
+
+    /* A list of entity types that will be processed */
+    protected List<String> entityTypes;
+    protected static final String ENTITY_TYPE_DEFAULT = "Publication";
+
+    /* A list of values the type might have */
+    protected List<String> trackerTypeMetadataValues;
+
+    /* The base url of the tracker */
+    protected String baseUrl;
+
+    protected String trackerUrlVersion;
+
+    protected static final String ITEM_VIEW = "Investigation";
+    protected static final String BITSTREAM_DOWNLOAD = "Request";
+
+    protected final static String UTF_8 = CharEncoding.UTF_8;
+
+    protected ConfigurationService configurationService = DSpaceServicesFactory.getInstance().getConfigurationService();
+    protected EntityTypeService entityTypeService = ContentServiceFactory.getInstance().getEntityTypeService();
+    protected EntityService entityService = ContentServiceFactory.getInstance().getEntityService();
+    protected ItemService itemService = ContentServiceFactory.getInstance().getItemService();
+    protected OpenUrlService openUrlService = OpenURLTrackerLoggerServiceFactory.getInstance().getOpenUrlService();
```

Comment:
These services should all be private.  You should not allow service sharing between classes, as all of these Services can be reloaded globally where needed.